File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-15/automation/2026-07-15-control-evidence-approval-record-validation-tool.ts
/** Synthetic AC-27 validation helper for p03-full-001838; no network or file access. */
const organization = "Cobalt Harbor Systems";
const workingPaperContext = { stage: "remediation-validation", evidenceCard: "EV-184-15", reviewLane: "PRIV-15", verificationPass: 2, workingPaperSequence: 1838, spineIds: ["org.cobalt-harbor", "control.approval-dual", "rel.ac27.evidence-184", "date.2026-07-15", "term.no-kio-claim"], rationale: "The control-owner remediation and retention disposition were validated as defensive governance evidence.", observation: "Approval separation records preparer and reviewer roles for the synthetic exception disposition.", canonicalValues: ["AC-27 requires preparer and reviewer role separation for each exception disposition."] };
type EvidenceRecord = { control_id: string; case_id: string; risk_id: string; evidence_set: number; reviewed: number; timely_attestations: number; residual_risk: string };
const expected: EvidenceRecord = { control_id: "AC-27", case_id: "CASE-260713-184", risk_id: "RISK-031", evidence_set: 184, reviewed: 48, timely_attestations: 47, residual_risk: "Medium" };
export function validate(record: EvidenceRecord): string[] { return Object.keys(expected).filter((key) => record[key as keyof EvidenceRecord] !== expected[key as keyof EvidenceRecord]); }
void organization; void workingPaperContext;

File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-15/automation/2026-07-15-control-evidence-approval-record-validation-tool.rs
// Synthetic AC-27 validation helper for p03-full-001834; no network or file access.
// stage=remediation-validation; card=EV-184-11; lane=PRIV-11; pass=9; working_paper=1834; spine=org.cobalt-harbor, evidence.ev-184-19, rel.ac27.evidence-184, date.2026-07-15, term.no-kio-claim
// The control-owner remediation and retention disposition were validated as defensive governance evidence. Approval separation records preparer and reviewer roles for the synthetic exception disposition.
const ORGANIZATION: &str = "Cobalt Harbor Systems";
pub fn valid(control_id: &str, case_id: &str, risk_id: &str, evidence_set: u32, reviewed: u32, timely: u32, residual: &str) -> bool {
    let _organization = ORGANIZATION;
    control_id == "AC-27" && case_id == "CASE-260713-184" && risk_id == "RISK-031" && evidence_set == 184 && reviewed == 48 && timely == 47 && residual == "Medium"
}
